experimental/casemapping/src/provider/exception_helpers.rs

```diff
@@ -0,0 +1,241 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains various types for the header part of casemapping exception data
+//!
+//! This is both used in datagen to decode ICU4C's data, and natively in ICU4X's
+//! own data model.
+//!
+//! [`ExceptionBits`] is the bag of bits associated with exceptions, and [`SlotPresence`]
+//! marks the presence or absence of various "slots" in a given exception.
+//!
+//! The `exceptions_builder` module of this crate handles decoding ICU4C data using the exception
+//! header, and [`crate::provider::exceptions`] handles.
+
+use crate::provider::data::{DotType, MappingKind};
+use zerovec::ule::{AsULE, ULE};
+
+/// A bunch of bits associated with each exception.
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+pub struct ExceptionBits {
```

Comment:
Thought (optional): It seems okay to make this a reflexive ULE type. Not sure if there's a performance difference between expanding the bitmask to a bool struct like this.